### ISSUE TITLE
Cross compilation to arm64 with clang-cl

### ISSUE THREAD

**Issue by mstorsjo** [mesonbuild/meson#7751]
I'm trying to cross compile with clang-cl to arm64. To achieve this, I'm using a cross file that looks like this:
```
[binaries]
c = 'clang-cl'
cpp = 'clang-cl'
ar = 'llvm-lib'

[properties]
c_args = ['--target=aarch64-win32-msvc']
cpp_args = ['--target=aarch64-win32-msvc']
```
Meson itself runs successfully, but builds later fail when running a command like `llvm-lib /NOLOGO /MACHINE:x64 ...`.

This is because meson tries to identify the target architecture, but does this only by running `clang-cl --version` and parses the output `Target:` line. If the arguments from `c_args` were to be included (this is in environment.py, _detect_c_or_cpp_compiler, in the `CL.EXE COMPATIBILITY` section), it would find the right architecture, but now it thinks it's targeting x86_64 because that's the compiler's default target.

I could try to work around this by adding a frontend wrapper to clang-cl that always passes the necessary `--target` option to clang-cl and setting this as the c compiler name in the cross file, but then this new command name needs to be added to the list of cl/clang-cl compiler names at the start of _detect_c_or_cpp_compiler.

**Comment by jon-turney** [comment]
Is this closed by PR #7750?

**Comment by mstorsjo** [comment]
> Is this closed by PR #7750?

No, it isn't. (To get to the issue that #7750 fixed, you'd either have to run clang-cl on an aarch64 host system, or hack around this issue in meson.)

Or put another way: This issue isn't entirely obvious how it's best to fix, so I didn't make a PR - but once it's fixed, whichever way, you'd run into that other issue, which was trivial to fix.

**Comment by jon-turney** [comment]
Ah right, I see the difference now.  Hmmm.

Also, if the cpu was specified in the cross-file, that should perhaps be controlling for the /MACHINE option, but probably isn't.

**Comment by jon-turney** [comment]
Does this work?

```
[binaries]
c = ['clang-cl', '--target=aarch64-win32-msvc']
```

**Comment by mstorsjo** [comment]
> Ah right, I see the difference now. Hmmm.
> 
> Also, if the cpu was specified in the cross-file, that should perhaps be controlling for the /MACHINE option, but probably isn't.

Indeed, that'd maybe be intuitive. This codepath for detecting it still is needed for cases when the compiler is implied otherwise, without using a cross file.


> Does this work?
> 
> ```
> [binaries]
> c = ['clang-cl', '--target=aarch64-win32-msvc']
> ```


Oh, it does - thanks! I hadn't realized that these could be arrays. So with that, and the fix from #7750, it does seem like it works with current master.
